src/cinder/Triangulate.cpp

```diff
@@ -108,22 +108,87 @@ void Triangulator::addPolyLine( const vec2 *points, size_t numPoints )
 TriMesh Triangulator::calcMesh( Winding winding )
 {
 	TriMesh result( TriMesh::Format().positions( 2 ) );
-	
+
 	tessTesselate( mTess.get(), (int)winding, TESS_POLYGONS, 3, 2, 0 );
 	result.appendPositions( (vec2*)tessGetVertices( mTess.get() ), tessGetVertexCount( mTess.get() ) );
 	result.appendIndices( (uint32_t*)( tessGetElements( mTess.get() ) ), tessGetElementCount( mTess.get() ) * 3 );
-	
+
 	return result;
 }
 
 TriMeshRef Triangulator::createMesh( Winding winding )
 {
 	TriMeshRef result = make_shared<TriMesh>( TriMesh::Format().positions( 2 ) );
-	
+
 	tessTesselate( mTess.get(), (int)winding, TESS_POLYGONS, 3, 2, 0 );
 	result->appendPositions( (vec2*)tessGetVertices( mTess.get() ), tessGetVertexCount( mTess.get() ) );
 	result->appendIndices( (uint32_t*)( tessGetElements( mTess.get() ) ), tessGetElementCount( mTess.get() ) * 3 );
-	
+
+	return result;
+}
+
+
+// * * *
+
+
+Triangulator3d::Triangulator3d( const PolyLine3f &polyLine )
+{
+	allocate();
+	addPolyLine( polyLine );
+}
+
+Triangulator3d::Triangulator3d()
+{
+	allocate();
+}
+
+void Triangulator3d::allocate()
+{
+	mAllocated = 0;
+
+	TESSalloc ma;
+	memset( &ma, 0, sizeof(ma) );
+	ma.memalloc = stdAlloc;
+	ma.memfree = stdFree;
+	ma.userData = (void*)&mAllocated;
+	ma.extraVertices = 2560; // realloc not provided, allow 256 extra vertices.
+
+	mTess = shared_ptr<TESStesselator>( tessNewTess( &ma ), tessDeleteTess );
+	if( ! mTess )
+		throw Triangulator::Exception();
+}
+
+void Triangulator3d::addPolyLine( const PolyLine3f &polyLine )
+{
+	if( polyLine.size() > 0 )
+		tessAddContour( mTess.get(), 3, &polyLine.getPoints()[0], sizeof(vec3), (int)polyLine.size() );
+}
+
+void Triangulator3d::addPolyLine( const vec3 *points, size_t numPoints )
+{
+	if( numPoints > 0 )
+		tessAddContour( mTess.get(), 3, &points, sizeof(vec3), (int)numPoints );
```

Comment:
Is the `&` in front of `points` correct here? I don't think we want a pointer to a pointer here